### PR TITLE
Add restart-sec option to systemd generate command

### PR DESCRIPTION
**Pull request by machacekondra**
#### What this PR does / why we need it:
This PR add new parameter called `restart-sec` to `podman generate systemd` command. This parameter is used to set `RestartSec` parameter in `[Service]` section of the pod unit in systemd. The `RestartSec` configures the time to sleep before restarting a service. Some users may need to set this variable to make sure the service is controlled by default mechanism.

#### How to verify it
When running following command:
```bash
podman generate systemd --restart-sec 15 pod_name
```
You should see in the pod unit [Service] section `RestartSec=15`.

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM, thank you for contributing

@Luap99 PTAL

**Comment by rhatdan** [comment]
In your tests you never actually specify the --restart-sec 15 call?

**Comment by machacekondra** [comment]
> In your tests you never actually specify the --restart-sec 15 call?

Right, thanks, I will add also command line test.

**Comment by vrothberg** [comment]
Good idea.  @machacekondra, please extend the tests in `./e2e/tests/`

File: test/e2e/ginkgo-node-2.log [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Running Suite: Libpod Suite
```
Comment (vrothberg) [inline test/e2e/ginkgo-node-2.log:1]: Can you remove this and the other log file from the PR?

**Review by vrothberg — COMMENTED** [review]
Once the two log files are removed, LGTM.

Thanks!

File: test/e2e/ginkgo-node-2.log (hunk 1 above)
Comment (machacekondra) [reply test/e2e/ginkgo-node-2.log]: Done.

**Comment by rhatdan** [comment]
Great job @machacekondra Thanks.
/lgtm
/approve

**Pull request merged**